cpp/src/arrow/compute/exec/accumulation_queue.h

```diff
@@ -42,9 +45,11 @@ class AccumulationQueue {
 
   void Concatenate(AccumulationQueue&& that);
   void InsertBatch(ExecBatch batch);
+  void InsertAt(ExecBatch batch, size_t idx);
```

Comment:
If I understand this method correctly it replaces the batch at index `idx`.  That was a little surprising since I think of `AccumulationQueue` as a vector and `std::vector` has an `insert` method that pushes elements to the right.  Can you change this to a name like `Replace` or `Set`?  Or at least add a brief doc comment.

Reply:
Renamed it to `SetBatch`.